src/Sarif.UnitTests/Core/StackTests.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Linq;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    [TestClass]
+    public class StackTests
+    {
+        [TestMethod]
+        public void Stack_CreateFromStackTrace()
+        {
+            var dotNetStack = new StackTrace();
+            Stack stack = Stack.Create(dotNetStack);
+
+            // The .NET StackTrace.ToString() override must preserve a trailing NewLine
+            // for compatibility reasons. We do not retain this behavior in ToString()
+            // but provide a facility for adding the trailing NewLine
+            Assert.AreEqual(dotNetStack.ToString(), stack.ToString(StackFormat.TrailingNewLine));
+        }
+
+        [TestMethod]
+        public void Stack_CreateFromException()
+        {
+            bool caughtException = false;
+            try
+            {
+                File.Create(Path.GetInvalidFileNameChars()[0].ToString(), 0);
+            }
+            catch (ArgumentException exception)
+            {
+                IEnumerable<Stack> stacks = Stack.Create(exception);
+                Assert.AreEqual(1, stacks.Count());
+                Assert.AreEqual(exception.StackTrace, stacks.First().ToString());
+                caughtException = true;
+            }
+            Assert.IsTrue(caughtException);
+        }
+
+        [TestMethod]
+        public void Stack_CreateFromExceptionWithInnerException()
+        {
+            bool caughtException = false;
+            try
+            {
+                File.Create(Path.GetInvalidFileNameChars()[0].ToString(), 0);
+            }
+            catch (ArgumentException exception)
+            {
+                Exception containerException = new InvalidOperationException("test exception", exception);
+
+                IEnumerable<Stack> stacks = Stack.Create(containerException);
+                Assert.AreEqual(2, stacks.Count());
```

Comment:
Larry, in re: using an [], I will stick in this case with IEnumerable. The reason is that the method itself doesn't know how many stacks it will return and so uses a List internally to build up the return value. I don't think it makes sense to take the cost upfront of allocating another return value when the list itself can be returned. It's not clear that consumers would need to convert to an array (a simple enumeration over the values makes sense). If it came to it, the helper could simply return the list, there's no real benefit to the helper to return a fixed array.